proxy_await.c

```diff
@@ -259,7 +259,11 @@ int mcplib_await_run(conn *c, mc_resp *resp, lua_State *L, int coro_ref) {
     const char *key = MCP_PARSER_KEY(rq->pr);
     size_t len = rq->pr.klen;
     int n = 0;
-    bool await_first = true;
+    // TODO (v3) await_first is used as a marker for upping the "wait for
+    // IO's" queue count, which means we need to force it off if we're in
+    // background mode, else we would accidentally wait for a response anyway.
+    // This note is for finding a less convoluted method for this.
+    bool await_first = (aw->type == AWAIT_BACKGROUND) ? false : true;
```

Comment:
note to self: I think this line can be removed, since we're checking for an AWAIT_BACKGROUND object before await_first, and they're handled the same now.

Reply:
Ah nevermind I see it again. it would increase the `q->count` twice.

guessing I should have the IO objects have an "increase the q->count" flag and just blindly add it to q->count, removing all of this test logic on submit_cb. even better might be adding a "next_count" counter to the IO queue and have callers tick that directly. Then copy or add it to q->count on the next submit. You can't kick q->count directly because it causes the system to never resume.

think that's a separate change.